File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  result = []
  split = str.split("")
  counter = split.count
  while counter >= 0
    result << split[counter]
    counter -= 1
  end
  result.join("")
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
  puts "Expecting: 'tset'"
  puts "=>", reverse_string('test')
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let result = []
  split = str.split("")
  for (i = split.length; i >= 0; i--) {
    result.push(split[i])
  }
  return result.join("")
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  // my tests
  console.log("Expecting: 'tset'")
  console.log("=>", reverseString("test"))
}

module.exports = reverseString;

// Please add your pseudocode to this file
  // initialize empty array called results
  // split string into array
  // iterate through array starting at last element
  // push each element into results array
  // return results array joined at commas

// And a written explanation of your solution
